### ISSUE TITLE
k-epsilon model

### ISSUE THREAD

**Issue by katyhuff** [arfc/moltres#57]
was accidentally previously filed in the wrong repo (https://github.com/arfc/publications/issues/44).

From @lindsayad :

The k-epsilon turbulence model was never finished since it really required SUPG stabilization to be relevant. Now, SUPG and PSPG are implemented in the navier-stokes module. So the k-epsilon model could be finished if someone wanted to work on it. It's very close to completion; the major piece remaining are the wall functions.

While Dr. Lindsay is not able to work on this directly, consult with @lindsayad to get appropriate initial direction.

**Comment by lcarasik** [comment]
What's y'all intended purpose here for the k-e model?

**Comment by katyhuff** [comment]
Thanks for your interest @lcarasik ! Ideally, this tool should as generally as possible support turbulent or near-turbulent flow regimes in liquid fueled MSRs. This k-epsilon feature is not intended for a specific problem, but rather to support realistic simulation of the broad class of liquid fueled MSRs. Natural convection is a significant component of the flow in MSRs. But for ordinary operation, the flow is driven primarily by the pumps. To us, k-epsilon seemed the simplest appropriate model for this scenario, but any advice is welcome!

**Comment by lcarasik** [comment]
I like the approach but I would warn you against trying to use it in transitional regimes. It might be worth looking into transitional RANS modeling if you get close. 

Is this 2-D? Or 3-D?

**Comment by gridley** [comment]
Hey Lane,

I don’t think there’s a particular application of the TH in mind at the moment. Most of the prior work has been trying to reproduce MSRE results, which had a tiny Re=900 or so.

Going up to full-power liquid-fueled MSRs, the Reynolds number seems to be very turbulent, so transitional stuff probably isn’t a huge worry at the moment.

What’s the difference in 2D and 3D?

**Comment by lcarasik** [comment]
Hey Gavin,

I assume y'all are just using N-S (no RANS) for the MSRE work?

It is still worth pointing out that if you get into transitional regimes (anything below 3000 for internal flows), you're going to see issues with standard RANS approaches.

Turbulence and transitional behavior is inherently 3-D phenomena. You won't capture a lot in 2-D that you will in 3-D. Though it may not be important for y'all's work.

**Comment by katyhuff** [comment]
Just for the record, this is false:

> I don’t think there’s a particular application of the TH in mind at the moment. 

:)

**Comment by gridley** [comment]
Which in particular then? Sorry, not that I was aware of, haha.

Gavin Ridley

**Comment by katyhuff** [comment]
Depends on which of my proposals get funded. :) 

In seriousness though, a number of designs in the open moltres issues (some of which my current graduate students are pursuing) as well as MSRE itself (as noted in the review of our paper) could be expected to reach near turbulent regimes during design basis accident transients.

**Comment by lcarasik** [comment]
It really sounds like you'd want someone to take a look at transition RANS after implementation of k-e!

**Comment by gridley** [comment]
Oh haha, nice! Out of curiosity, if it's not classified info, how could the MSRE go turbulent? The only thing I can imagine would be somehow blocking half of the channels. Aside from that, the reviewer's claim of near turbulence didn't agree with @lindsayad's calculation at all (which I trust more). The pump ran at 80% power since the heat transfer calcs were slightly off, which led them to not run at the full 10 MW(th). So Re could max out around 1200, neglecting half of the channels being blocked.

**Comment by katyhuff** [comment]
Yes, @lcarasik , particularly someone with your expertise! 

I said 'near turbulent regimes during design basis accident transients' not 'turbulent in a steady state scenario.'  In any case, there are plenty of examples of potentially laminar-to-turbulent even in not-far-off-normal MSRE experience. To quote the following about the MSRE experience.... "The regular pattern of the graphite support grid is discontinued near the centerline of the core where the control rods and the surveillance specimin holder are located. This allows greater salt velocities past these components for cooling.... The reynolds number of these channels is in the order of 3000 and is therefore in the transitional region." I'm not sure our published model took into account these larger central channels, as many don't. 

Additionally, not to let any cats out of any bags, but some atypical dynamic operation of some of these reactors (intentional or otherwise) might be constrained by the impact of rapid power oscillations on the stability of the velocity field. All of that is to say I don't think we can ignore near-turbulent behavior for interesting transients. 

Anyway, the geometry varies all over the place among the proposed designs. Some have pipes, some grids, some plates, some pools. The tool shouldn't be constrained to laminar regimes.

**Comment by gridley** [comment]
I agree! Laminar is boring (comparatively). This should be exciting.

**Comment by smpark7** [comment]
I'm working towards implementing and demonstrating this model for MSFRs modeling. I noticed that @lindsayad had already coded much of the work in the ```INSMomentumKEpsilon``` and ```INSK``` kernels (although ```INSK``` had been moved out of Moltres in a previous commit). May I inquire how much has been completed and where I should start?

**Comment by lcarasik** [comment]
Just a quick drop in comment, please make sure the flow is fully turbulent if you apply it! If it isn't fully turbulent, you'll applying the turbulence model outside of it's applicability.

**Comment by katyhuff** [comment]
> May I inquire how much has been completed and where I should start?
Let's chat in person?....

**Comment by katyhuff** [comment]
@smpark7 let's chat in person about your question:
> May I inquire how much has been completed and where I should start?
